### PR TITLE
Ensure people don't overwrite staffcp urls

### PR DESCRIPTION
**Pull request by Derkades**
I still need to add a proper message

### CODE REVIEW

File: modules/Core/pages/panel/pages.php [hunk 1]
```diff
@@ -23,6 +23,48 @@
 // Load modules + template
 Module::loadPage($user, $pages, $cache, $smarty, [$navigation, $cc_nav, $staffcp_nav], $widgets, $template);
 
+function validate_page_form() {
```
Comment (Aberdeener) [inline modules/Core/pages/panel/pages.php:26]: While I like the idea of removing duplicate validation checks, I don't know if this is the most elegant way we can do it...

Comment (Derkades) [reply modules/Core/pages/panel/pages.php:26]: Any other ideas? A collection of "form validators" in a separate directory somewhere?

Comment (Aberdeener) [reply modules/Core/pages/panel/pages.php:26]: Perhaps - that seems good.
`FormValidator` class which contains a bunch of static functions seems alright. Ideally we use something like https://laravel.com/docs/9.x/validation#form-request-validation but can't get too ahead of ourselves 😉

File: modules/Core/pages/panel/pages.php [hunk 2]
```diff
@@ -23,6 +23,48 @@
...
+        'page_title' => [
+            Validate::REQUIRED => true,
+            Validate::MIN => 2,
+            Validate::MAX => 255
+        ],
+        'page_url' => [
+            Validate::REQUIRED => true,
+            Validate::MIN => 2,
+            Validate::MAX => 255,
+            Validate::NOT_START_WITH => '/panel'
```
Comment (samerton) [inline modules/Core/pages/panel/pages.php:38]: We need to consider subdirectory installations here

Comment (partydragen) [reply modules/Core/pages/panel/pages.php:38]: Unfriendly urls don't even require to begin with / , so should make sure page path always begins with a /

File: modules/Core/pages/panel/pages.php (hunk 1 above)
Comment (Derkades) [reply modules/Core/pages/panel/pages.php:26]: should that be included in this PR, or can it be done later?

File: modules/Core/pages/panel/pages.php (hunk 2 above)
Comment (Derkades) [reply modules/Core/pages/panel/pages.php]: I believe the entered custom page path wouldn't contain subdirectories, so I don't think we need to do anything special to support subdirectory installations?

Comment (partydragen) [reply modules/Core/pages/panel/pages.php]: yea you're right, we don't enter subdir there

File: modules/Core/pages/panel/pages.php (hunk 1 above)
Comment (Aberdeener) [reply modules/Core/pages/panel/pages.php:26]: let's do that later:tm:

for now, let's keep the repeated validation in the file.

**Pull request merged**